hummingbot/connector/utilities/oms_connector/oms_connector_exchange.py

```diff
@@ -197,12 +201,27 @@ async def _place_cancel(self, order_id: str, tracked_order: InFlightOrder):
             data=params,
             is_auth_required=True,
         )
+        self.logger().debug(f"Cancelation result of {tracked_order.client_order_id} at {start_ts}: {cancel_result}")
+        cancel_success = False
         if cancel_result.get(CONSTANTS.ERROR_CODE_FIELD):
-            if cancel_result[CONSTANTS.ERROR_MSG_FIELD] == CONSTANTS.RESOURCE_NOT_FOUND_MSG:
-                await self._order_tracker.process_order_not_found(order_id)
+            if cancel_result[CONSTANTS.ERROR_CODE_FIELD] == CONSTANTS.RESOURCE_NOT_FOUND_ERR_CODE:
+                self._order_not_found_on_cancel_record[order_id] += 1
+                if self._order_not_found_on_cancel_record[order_id] >= CONSTANTS.MAX_ORDER_NOT_FOUND_ON_CANCEL:
+                    cancel_success = True
```

Comment:
Why are we using here and ad hoc logic to keep track of the cancellation error? Should we not use the logic in the ClientOrderTracker class?

Reply:
If the order has been cancelled asynchronously before this cancellation is sent out, the exchange will return a "resource not found" error. If we feed that to the tracker, after 10 such errors, it will consider the order as failed. Because of this, I went with the approach here where a resource after 2 tries is taken as a successful cancellation. The status update logic still uses the default tracker behaviour however.

Reply:
Discussed offline. I will change this to use the tracker logic because the previous fixes to cancellation tracking should negate the need of this ad-hoc logic.